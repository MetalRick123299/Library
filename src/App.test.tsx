import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';
import { IBookItem } from './BookItem';

describe('Main App', () => {
  const book: IBookItem = {
    title: 'Harry Potter',
    author: 'Not J.K Rowling',
    pagesRead: 45,
    totalPages: 586,
  };

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let app: any;
  beforeEach(() => {
    app = render(<App />);
  });

  it('App Structure', () => {
    const library = screen.getByRole('heading', { level: 1 });
    const addButton = screen.getByRole('button', { name: 'Add Book' });
    const signInButton = screen.getByRole('button', { name: /sign in/i });

    expect(library.innerHTML).toBe('Library');
    expect(addButton.innerHTML).toBe('Add Book');
    expect(signInButton.innerHTML).toBe('Sign In');
  });
  it('Snap Shot', () => {
    expect(app).toMatchSnapshot();
  });

  it('Add Book Modal Appears / Disappears on Click', () => {
    let modal = screen.queryByRole('form', { name: 'Add Book Form' });
    expect(modal).toBe(null);

    const addBookButton = screen.getByRole('button', { name: 'Add Book' });
    fireEvent.click(addBookButton);

    modal = screen.queryByRole('form', { name: 'Add Book Form' });
    expect(modal).not.toBe(null);

    const closeButton = screen.getByRole('button', {
      name: 'Close Form Button',
    });
    fireEvent.click(closeButton);
    modal = screen.queryByRole('form', { name: 'Add Book Form' });
    expect(modal).toBe(null);
  });
  it('Adding a Book', () => {
    const addBookButton = screen.getByRole('button', { name: 'Add Book' });
    fireEvent.click(addBookButton);

    const titleInput = screen.getByRole('textbox', { name: /title/i });
    const authorInput = screen.getByRole('textbox', { name: /author/i });
    const pagesReadInput = screen.getByRole('spinbutton', {
      name: /Pages Read/i,
    });
    const totalPagesInput = screen.getByRole('spinbutton', {
      name: /Total Pages/i,
    });

    fireEvent.change(titleInput, { target: { value: book.title } });
    fireEvent.change(authorInput, { target: { value: book.author } });
    fireEvent.change(pagesReadInput, { target: { value: book.pagesRead } });
    fireEvent.change(totalPagesInput, { target: { value: book.totalPages } });

    const addBookSubmit = screen.getByRole('button', {
      name: 'Add Book Submit',
    });
    fireEvent.click(addBookSubmit);

    const title = screen.getByRole('heading', { level: 2, name: book.title });
    const author = screen.getByText(book.author);
    const pagesElement = screen.getByText(
      `${book.pagesRead} / ${book.totalPages}`
    );
    const progress = screen.getByText(
      `${parseFloat((book.pagesRead / book.totalPages).toFixed(4)) * 100}%`
    );

    expect(title.innerHTML).toBe(book.title);
    expect(author.innerHTML).toBe(book.author);
    expect(pagesElement.innerHTML).toBe(
      `${book.pagesRead} / ${book.totalPages}`
    );
    expect(progress).toBe(progress);
  });

  it.todo('No Same Named Books', () => {});
  it('Deleting a Book', () => {
    const addBookButton = screen.getByRole('button', { name: 'Add Book' });
    fireEvent.click(addBookButton);

    const titleInput = screen.getByRole('textbox', { name: /title/i });
    const authorInput = screen.getByRole('textbox', { name: /author/i });
    const pagesReadInput = screen.getByRole('spinbutton', {
      name: /Pages Read/i,
    });
    const totalPagesInput = screen.getByRole('spinbutton', {
      name: /Total Pages/i,
    });

    fireEvent.change(titleInput, { target: { value: book.title } });
    fireEvent.change(authorInput, { target: { value: book.author } });
    fireEvent.change(pagesReadInput, { target: { value: book.pagesRead } });
    fireEvent.change(totalPagesInput, { target: { value: book.totalPages } });

    const addBookSubmit = screen.getByRole('button', {
      name: 'Add Book Submit',
    });
    fireEvent.click(addBookSubmit);

    const title = screen.getByRole('heading', { level: 2, name: book.title });
    const deleteButtons = screen.getAllByRole('button', { name: /delete/i });
    const idx = deleteButtons.findIndex(
      (item) =>
        item.parentElement?.firstElementChild?.innerHTML === title.innerHTML
    );
    fireEvent.click(deleteButtons[idx]);

    expect(screen.queryByRole('heading', { level: 2, name: book.title })).toBe(
      null
    );
  });

  // Do => 12/01/22 -> Firebase Update
  it.todo('Signing In', () => {});
  it.todo('Signing Out', () => {});
});
