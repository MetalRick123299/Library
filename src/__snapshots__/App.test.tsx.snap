// Vitest Snapshot v1

exports[`Main App > Snap Shot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="min-h-full text-white bg-primary-bg"
      >
        <div
          aria-hidden="true"
          class="fixed flex flex-col justify-center items-center h-screen w-screen bg-black/60 z-10 text-xl transition-all scale-0"
        >
          <form
            action=""
            aria-label="Add Book Form"
            class="flex flex-col items-center gap-3 bg-primary-bg p-5 rounded-xl border-primary-nav border-8"
          >
            <label
              class=""
            >
              Search Book
              <input
                class="bg-primary-item block rounded-md px-2 py-1 mt-1"
                name="search Book"
                type="text"
              />
            </label>
            <div
              class="text-2xl -mb-3 font-semibold"
            >
              OR
            </div>
            <label
              class=""
            >
              Title
              <input
                class="bg-primary-item block rounded-md px-2 py-1 mt-1"
                name="title"
                required=""
                type="text"
                value=""
              />
            </label>
            <label
              class=""
            >
              Author
              <input
                class="bg-primary-item block rounded-md px-2 py-1 mt-1"
                name="author"
                required=""
                type="text"
                value=""
              />
            </label>
            <label
              class=""
            >
              Pages Read
              <input
                class="bg-primary-item block rounded-md px-2 py-1 mt-1"
                name="pages Read"
                required=""
                type="number"
                value="0"
              />
            </label>
            <label
              class=""
            >
              Total Pages
              <input
                class="bg-primary-item block rounded-md px-2 py-1 mt-1"
                name="total Pages"
                required=""
                type="number"
                value="0"
              />
            </label>
            <div
              class="flex items-start gap-3 bg-primary-item p-3 rounded-xl"
            >
              <button
                aria-label="Add Book Submit"
                class=""
                type="submit"
              >
                Add Book
              </button>
              <svg
                aria-hidden="false"
                aria-label="Close Form Button"
                class="btn inline !h-7"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM8.28 7.22a.75.75 0 00-1.06 1.06L8.94 10l-1.72 1.72a.75.75 0 101.06 1.06L10 11.06l1.72 1.72a.75.75 0 101.06-1.06L11.06 10l1.72-1.72a.75.75 0 00-1.06-1.06L10 8.94 8.28 7.22z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </form>
        </div>
        <nav
          class="bg-primary-nav h-24 grid grid-cols-3 place-items-center fixed inset-0 z-[2]"
        >
          <h1
            class="text-4xl"
          >
            Library
          </h1>
          <button
            class="mx-auto text-3xl border-2 border-primary-bg px-6 py-3 rounded-full shadow-xl active:shadow-none active:translate-y-1 transition-all whitespace-nowrap"
            type="button"
          >
            Add Book
          </button>
          <div
            class="flex gap-3"
          >
            <button
              class="text-4xl"
              type="button"
            >
              Sign In
            </button>
            <img
              alt="Profile"
              class="rounded-full hidden"
              src="https://via.placeholder.com/48"
            />
            <button
              class="text-2xl hidden"
              type="button"
            >
              Sign Out
            </button>
          </div>
        </nav>
        <div
          class="pt-28 flex gap-10 flex-wrap justify-center items-center"
        >
          <div
            class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
          >
            <h2
              class=""
            >
              Book Title 1
            </h2>
            <span
              class=""
            >
              Maybe JK
            </span>
            <span
              class=""
            >
              256
               / 
              586
            </span>
            <div
              class="flex w-full items-center justify-between"
            >
              <svg
                aria-hidden="false"
                aria-label="Down Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
              >
                43.69
                %
                <div
                  class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                  style="width: 43.69%;"
                />
              </div>
              <svg
                aria-hidden="false"
                aria-label="Up Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <button
              class="bg-red-500 self-stretch rounded-lg p-1"
              type="button"
            >
              Delete
            </button>
          </div>
          <div
            class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
          >
            <h2
              class=""
            >
              Book Title 2
            </h2>
            <span
              class=""
            >
              Maybe JK
            </span>
            <span
              class=""
            >
              0
               / 
              586
            </span>
            <div
              class="flex w-full items-center justify-between"
            >
              <svg
                aria-hidden="false"
                aria-label="Down Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
              >
                0.00
                %
                <div
                  class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                  style="width: 0.00%;"
                />
              </div>
              <svg
                aria-hidden="false"
                aria-label="Up Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <button
              class="bg-red-500 self-stretch rounded-lg p-1"
              type="button"
            >
              Delete
            </button>
          </div>
          <div
            class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
          >
            <h2
              class=""
            >
              Book Title 3
            </h2>
            <span
              class=""
            >
              Maybe JK
            </span>
            <span
              class=""
            >
              256
               / 
              586
            </span>
            <div
              class="flex w-full items-center justify-between"
            >
              <svg
                aria-hidden="false"
                aria-label="Down Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
              >
                43.69
                %
                <div
                  class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                  style="width: 43.69%;"
                />
              </div>
              <svg
                aria-hidden="false"
                aria-label="Up Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <button
              class="bg-red-500 self-stretch rounded-lg p-1"
              type="button"
            >
              Delete
            </button>
          </div>
          <div
            class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
          >
            <h2
              class=""
            >
              Book Title 4
            </h2>
            <span
              class=""
            >
              Maybe JK
            </span>
            <span
              class=""
            >
              586
               / 
              586
            </span>
            <div
              class="flex w-full items-center justify-between"
            >
              <svg
                aria-hidden="false"
                aria-label="Down Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
              >
                100.00
                %
                <div
                  class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                  style="width: 100.00%;"
                />
              </div>
              <svg
                aria-hidden="false"
                aria-label="Up Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <button
              class="bg-red-500 self-stretch rounded-lg p-1"
              type="button"
            >
              Delete
            </button>
          </div>
          <div
            class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
          >
            <h2
              class=""
            >
              Book Title 5
            </h2>
            <span
              class=""
            >
              Maybe JK
            </span>
            <span
              class=""
            >
              256
               / 
              586
            </span>
            <div
              class="flex w-full items-center justify-between"
            >
              <svg
                aria-hidden="false"
                aria-label="Down Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
              >
                43.69
                %
                <div
                  class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                  style="width: 43.69%;"
                />
              </div>
              <svg
                aria-hidden="false"
                aria-label="Up Arrow"
                class="btn"
                fill="currentColor"
                role="button"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <button
              class="bg-red-500 self-stretch rounded-lg p-1"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="min-h-full text-white bg-primary-bg"
    >
      <div
        aria-hidden="true"
        class="fixed flex flex-col justify-center items-center h-screen w-screen bg-black/60 z-10 text-xl transition-all scale-0"
      >
        <form
          action=""
          aria-label="Add Book Form"
          class="flex flex-col items-center gap-3 bg-primary-bg p-5 rounded-xl border-primary-nav border-8"
        >
          <label
            class=""
          >
            Search Book
            <input
              class="bg-primary-item block rounded-md px-2 py-1 mt-1"
              name="search Book"
              type="text"
            />
          </label>
          <div
            class="text-2xl -mb-3 font-semibold"
          >
            OR
          </div>
          <label
            class=""
          >
            Title
            <input
              class="bg-primary-item block rounded-md px-2 py-1 mt-1"
              name="title"
              required=""
              type="text"
              value=""
            />
          </label>
          <label
            class=""
          >
            Author
            <input
              class="bg-primary-item block rounded-md px-2 py-1 mt-1"
              name="author"
              required=""
              type="text"
              value=""
            />
          </label>
          <label
            class=""
          >
            Pages Read
            <input
              class="bg-primary-item block rounded-md px-2 py-1 mt-1"
              name="pages Read"
              required=""
              type="number"
              value="0"
            />
          </label>
          <label
            class=""
          >
            Total Pages
            <input
              class="bg-primary-item block rounded-md px-2 py-1 mt-1"
              name="total Pages"
              required=""
              type="number"
              value="0"
            />
          </label>
          <div
            class="flex items-start gap-3 bg-primary-item p-3 rounded-xl"
          >
            <button
              aria-label="Add Book Submit"
              class=""
              type="submit"
            >
              Add Book
            </button>
            <svg
              aria-hidden="false"
              aria-label="Close Form Button"
              class="btn inline !h-7"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zM8.28 7.22a.75.75 0 00-1.06 1.06L8.94 10l-1.72 1.72a.75.75 0 101.06 1.06L10 11.06l1.72 1.72a.75.75 0 101.06-1.06L11.06 10l1.72-1.72a.75.75 0 00-1.06-1.06L10 8.94 8.28 7.22z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </form>
      </div>
      <nav
        class="bg-primary-nav h-24 grid grid-cols-3 place-items-center fixed inset-0 z-[2]"
      >
        <h1
          class="text-4xl"
        >
          Library
        </h1>
        <button
          class="mx-auto text-3xl border-2 border-primary-bg px-6 py-3 rounded-full shadow-xl active:shadow-none active:translate-y-1 transition-all whitespace-nowrap"
          type="button"
        >
          Add Book
        </button>
        <div
          class="flex gap-3"
        >
          <button
            class="text-4xl"
            type="button"
          >
            Sign In
          </button>
          <img
            alt="Profile"
            class="rounded-full hidden"
            src="https://via.placeholder.com/48"
          />
          <button
            class="text-2xl hidden"
            type="button"
          >
            Sign Out
          </button>
        </div>
      </nav>
      <div
        class="pt-28 flex gap-10 flex-wrap justify-center items-center"
      >
        <div
          class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
        >
          <h2
            class=""
          >
            Book Title 1
          </h2>
          <span
            class=""
          >
            Maybe JK
          </span>
          <span
            class=""
          >
            256
             / 
            586
          </span>
          <div
            class="flex w-full items-center justify-between"
          >
            <svg
              aria-hidden="false"
              aria-label="Down Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
            >
              43.69
              %
              <div
                class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                style="width: 43.69%;"
              />
            </div>
            <svg
              aria-hidden="false"
              aria-label="Up Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <button
            class="bg-red-500 self-stretch rounded-lg p-1"
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
        >
          <h2
            class=""
          >
            Book Title 2
          </h2>
          <span
            class=""
          >
            Maybe JK
          </span>
          <span
            class=""
          >
            0
             / 
            586
          </span>
          <div
            class="flex w-full items-center justify-between"
          >
            <svg
              aria-hidden="false"
              aria-label="Down Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
            >
              0.00
              %
              <div
                class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                style="width: 0.00%;"
              />
            </div>
            <svg
              aria-hidden="false"
              aria-label="Up Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <button
            class="bg-red-500 self-stretch rounded-lg p-1"
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
        >
          <h2
            class=""
          >
            Book Title 3
          </h2>
          <span
            class=""
          >
            Maybe JK
          </span>
          <span
            class=""
          >
            256
             / 
            586
          </span>
          <div
            class="flex w-full items-center justify-between"
          >
            <svg
              aria-hidden="false"
              aria-label="Down Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
            >
              43.69
              %
              <div
                class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                style="width: 43.69%;"
              />
            </div>
            <svg
              aria-hidden="false"
              aria-label="Up Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <button
            class="bg-red-500 self-stretch rounded-lg p-1"
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
        >
          <h2
            class=""
          >
            Book Title 4
          </h2>
          <span
            class=""
          >
            Maybe JK
          </span>
          <span
            class=""
          >
            586
             / 
            586
          </span>
          <div
            class="flex w-full items-center justify-between"
          >
            <svg
              aria-hidden="false"
              aria-label="Down Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
            >
              100.00
              %
              <div
                class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                style="width: 100.00%;"
              />
            </div>
            <svg
              aria-hidden="false"
              aria-label="Up Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <button
            class="bg-red-500 self-stretch rounded-lg p-1"
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="bg-primary-item flex flex-col items-center gap-3 p-5 rounded-xl w-64 text-2xl"
        >
          <h2
            class=""
          >
            Book Title 5
          </h2>
          <span
            class=""
          >
            Maybe JK
          </span>
          <span
            class=""
          >
            256
             / 
            586
          </span>
          <div
            class="flex w-full items-center justify-between"
          >
            <svg
              aria-hidden="false"
              aria-label="Down Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="w-8/12 rounded-lg border-2 border-primary-bg text-center relative z-[1]"
            >
              43.69
              %
              <div
                class="h-full bg-primary-bg absolute inset-0 -z-[1] transition-all max-w-full"
                style="width: 43.69%;"
              />
            </div>
            <svg
              aria-hidden="false"
              aria-label="Up Arrow"
              class="btn"
              fill="currentColor"
              role="button"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <button
            class="bg-red-500 self-stretch rounded-lg p-1"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
